Author: Daniel F. Dickinson <cshored@thecshore.com>
Date:   Tue Jan 16 03:06:33 2018 -0500

    busybox: Fix compilation for musl on 32-bit archs

    At least with core-image-minimal busybox triggers the recently
    added off_t misdetection alert even when LFS is not enabled;
    this is due to a bug in busybox handling of the case...this
    patch should get upstremed.

    Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>
Index: busybox-1.24.1/include/libbb.h
===================================================================
--- busybox-1.24.1.orig/include/libbb.h
+++ busybox-1.24.1/include/libbb.h
@@ -252,6 +252,14 @@ typedef unsigned long long uoff_t;
 #  define STRTOOFF strtoull
 #  define OFF_FMT "ll"
 # endif
+/* scary. better ideas? (but do *test* them first!) */
+#define OFF_T_MAX  ((off_t)~((off_t)1 << (sizeof(off_t)*8-1)))
+/* Users report bionic to use 32-bit off_t even if LARGEFILE support is requested.
+ * We misdetected that. Don't let it build:
+ */
+struct BUG_off_t_size_is_misdetected {
+	char BUG_off_t_size_is_misdetected[sizeof(off_t) == sizeof(uoff_t) ? 1 : -1];
+};
 #else
 /* CONFIG_LFS is off */
 # if UINT_MAX == 0xffffffff
@@ -269,15 +277,8 @@ typedef unsigned long uoff_t;
 #  define STRTOOFF strtol
 #  define OFF_FMT "l"
 # endif
+#define OFF_T_MAX UINT_MAX
 #endif
-/* scary. better ideas? (but do *test* them first!) */
-#define OFF_T_MAX  ((off_t)~((off_t)1 << (sizeof(off_t)*8-1)))
-/* Users report bionic to use 32-bit off_t even if LARGEFILE support is requested.
- * We misdetected that. Don't let it build:
- */
-struct BUG_off_t_size_is_misdetected {
-	char BUG_off_t_size_is_misdetected[sizeof(off_t) == sizeof(uoff_t) ? 1 : -1];
-};
 
 /* Some useful definitions */
 #undef FALSE
